lib/base/utility.cpp

```diff
@@ -1779,7 +1779,7 @@ const char l_Utf8Replacement[] = "\xEF\xBF\xBD";
 
 String Utility::ValidateUTF8(const String& input)
 {
-	std::vector<char> output;
+	std::string output;
 	output.reserve(input.GetLength() * 3u);
```

Comment:
This allocates for the worst case where every character in the string has to be replaced, ...